localization.c

```diff
@@ -354,9 +354,54 @@ FillLangListProc(UNUSED HANDLE module, UNUSED PTSTR type, UNUSED PTSTR stringId,
     return TRUE;
 }
 
+BOOL 
+GetLaunchOnStartup()
+{
+	
+    TCHAR regPath[MAX_PATH], exePath[MAX_PATH];	
+    BOOL result = FALSE;
+    HKEY regkey;
+
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, _T("Software\\Microsoft\\Windows\\CurrentVersion\\Run"), 0, KEY_READ, &regkey) == ERROR_SUCCESS) {
+
+        if (GetRegistryValue(regkey, _T("OpenVPN-GUI"), regPath, MAX_PATH) &&
+            GetModuleFileName(NULL, exePath, MAX_PATH)) {
+            if (_tcsicmp(regPath, exePath) == 0)
+                result = TRUE;
+        }
+
+        RegCloseKey(regkey);
+
+    }
+	
+    return result;
+
+}
+
+void 
+SetLaunchOnStartup(BOOL value) 
+{
+
+    TCHAR exePath[MAX_PATH];
+    HKEY regkey;
+
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, _T("Software\\Microsoft\\Windows\\CurrentVersion\\Run"), 0, KEY_WRITE, &regkey) == ERROR_SUCCESS) {
+
+        if (value) {
+            if (GetModuleFileName(NULL, exePath, MAX_PATH)) 
+                SetRegistryValue(regkey, _T("OpenVPN-GUI"), exePath);
+        }
+        else 
+            RegDeleteValue(regkey, _T("OpenVPN-GUI"));            
+
+        RegCloseKey(regkey);
+
+    }
+
+}
```

Comment:
We no longer support non-unicode compilation so avoid TCHAR and _t functions. In any case prefer explicit character types in all new code as far as possible. That is, WCHAR or char (as appropriate) instead of TCHAR, explicitly use functions ending W for wide chars, L instead of T() etc. Also TCHAR in exisitng code may be assumed to mean WCHAR.